client/api_client.go

```diff
@@ -119,6 +126,23 @@ type OffChainAuthOption struct {
 	ShouldRegisterPubKey bool
 }
 
+// OffChainAuthOptionV2 - The optional configurations for off-chain-auth-v2.
+//
+// The OffChainAuthOptionV2 consists of a EdDSA private key and the domain where the EdDSA keys will be registered for.
+//
+// This auth mechanism is usually used in browser-based application.
+// That we support OffChainAuth configuration in go-sdk is to make the tests on off-chain-auth-v2 be convenient.
+type OffChainAuthOptionV2 struct {
```

Comment:
Why not reuse the OffChainAuthOption by adding new fields to differentiate between v1 and v2 versions?